javalib/src/main/scala/java/lang/StackTraceElement.scala

```diff
@@ -1,15 +1,91 @@
 package java.lang
 
-final class StackTraceElement(val getClassName: String,
-                              val getMethodName: String,
-                              val getFileName: String,
-                              val getLineNumber: Int) {
+import scala.scalanative.native.{
+  fromCString,
+  stackalloc,
+  string,
+  CChar,
+  CString,
+  CUnsignedLong,
+  Ptr
+}
+import scala.scalanative.runtime.{ByteArray, unwind}
+
+final class StackTraceElement private (_symbol: CString) {
+  var symbol: Array[Byte] = _
+
+  if (_symbol != null) {
+    symbol = new Array[Byte](256)
+    string.memcpy(symbol.asInstanceOf[ByteArray].at(0), _symbol, 256)
+  }
+
+  def this(className: String,
+           methodName: String,
+           fileName: String,
+           lineNumber: Int) = {
+    this(null)
+    if (className == null)
+      throw new NullPointerException("Declaring class is null")
 
-  if (getClassName == null)
-    throw new NullPointerException("Declaring class is null")
+    if (methodName == null)
+      throw new NullPointerException("Method name is null")
+
+    initDone = true
+    _className = className
+    _methodName = methodName
+    _fileName = fileName
+    _lineNumber = lineNumber
+  }
+
+  private var initDone: Boolean   = false
+  private var _className: String  = _
+  private var _methodName: String = _
+  private var _fileName: String   = _
+  private var _lineNumber: Int    = _
+
+  def getClassName: String = {
+    if (!initDone) initMembers()
+    _className
+  }
+
+  def getMethodName: String = {
+    if (!initDone) initMembers()
+    _methodName
+  }
+
+  def getFileName: String = {
+    if (!initDone) initMembers()
+    _fileName
+  }
+
+  def getLineNumber: Int = {
+    if (!initDone) initMembers()
+    _lineNumber
+  }
+
+  // symbol is "classname::methodName_T1_..._TN"
+  // where classname doesn't include colons and method name underscores.
+  private def initMembers(): Unit = {
+    val sym = fromCString(symbol.asInstanceOf[ByteArray].at(0))
+    val (className, methodName) =
+      sym.indexOf("::") match {
+        case -1 =>
+          ("<none>", sym)
+        case sep =>
+          sym.indexOf("_", sep) match {
+            case -1 =>
+              (sym.substring(0, sep), sym.substring(sep + 2))
+            case end =>
+              (sym.substring(0, sep), sym.substring(sep + 2, end))
+          }
+      }
 
-  if (getMethodName == null)
-    throw new NullPointerException("Method name is null")
+    initDone = true
+    _className = className
+    _methodName = methodName
+    _fileName = null
+    _lineNumber = 0
```

Comment:
As soon as you complete the initialization you can set `symbol` field to null as it's never going to be used again.